src/emmetHelper.ts

```diff
@@ -495,7 +495,11 @@ export function isAbbreviationValid(syntax: string, abbreviation: string): boole
 			return false;
 		}
 		if (abbreviation.includes('#')) {
-			return hexColorRegex.test(abbreviation) || propertyHexColorRegex.test(abbreviation);
+			if (abbreviation.startsWith('#')) {
+				return hexColorRegex.test(abbreviation);
+			} else if (commonlyUsedTags.includes(abbreviation.substring(0, abbreviation.indexOf('#')))) {
+				return false;
```

Comment:
Not sure what this case is for, considering we're in a `if (isStyleSheet(syntax))` block

Reply:
@rzhao271 The whole `if (abbreviation.includes('#'))` block was added as a fix for this https://github.com/microsoft/vscode/issues/56082 where emmet completions would show for id selectors and that's not desired. `propertyHexColorRegex` regex is too restrictive causing expressions like `bd1#s` not being expanded so the fix is to not show completions only for id selectors of the form `<element>#<id>`